File: projects/inflection/modules/dev/dev_py/ru/declension/run/parts/_prepare.py
from projects.inflection.modules.dev.dev_py import a
from projects.inflection.modules.dev.dev_py import mw
from projects.inflection.modules.dev.dev_py import tools as _

dev_prefix = 'User:Vitalik/'  # comment this on `prod` version


from ...run.parts.prepare import endings as endings
from ...run.parts.prepare import stress_apply as stress_apply


module = 'run.parts.prepare'  # local


@a.starts(module)
def prepare(func, i):  # export
    # INFO: Generates `.endings` and `.stems`

    # todo: logging info
    endings.get_endings(i)

    # todo: logging info
    i.parts.stems = dict()  # dict
    stress_apply.apply_stress_type(i)
    _.log_table(i.parts.stems, 'info.parts.stems')
    _.log_table(i.parts.endings, 'info.parts.endings')

    _.ends(module, func)
# end


# return export

File: projects/inflection/modules/prod/prod_lua/ru/declension/run/parts/_prepare.lua
local dev_prefix = ''
dev_prefix = 'User:Vitalik/'  -- comment this on `prod` version

local export = {}
local _ = require('Module:' .. dev_prefix .. 'inflection/tools')


local endings = require('Module:' .. dev_prefix .. 'inflection/ru/declension/run/parts/prepare/endings')  -- '..'
local stress_apply = require('Module:' .. dev_prefix .. 'inflection/ru/declension/run/parts/prepare/stress_apply')  -- '..'


local module = 'run.parts.prepare'


-- @starts
function export.prepare(i)
	func = "prepare"
	_.starts(module, func)

--	INFO: Generates `.endings` and `.stems`

	-- todo: logging information
	endings.get_endings(i)

	-- todo: logging information
	i.parts.stems = {}  -- dict
	stress_apply.apply_stress_type(i)
	_.log_table(i.parts.stems, 'i.parts.stems')
	_.log_table(i.parts.endings, 'i.parts.endings')

	_.ends(module, func)
end


return export
